### ISSUE TITLE
feat: Upgrade physics to Box2D

### ISSUE THREAD

**Issue by quinton-ashley** [marklovers/kaplay#252]
[p5play](https://p5play.org), [Construct 3](https://www.construct.net/en), [GDevelop](https://gdevelop.io/), and Unity all use the Box2D physics simulator.

Matter.js is not a good physics simulator and was probably only used in kaboom.js due to device limitations years ago.

**Comment by niceEli** [comment]
As of right now the physics is plugin based. It would be best to do this and make accurate physics part of engine itself at the same time.

**Comment by mflerackers** [comment]
No, physics needs to be plugin based, because we need to support multiple physics engines. Box2D (or PlanckJS or MatterJS) are not a good choice for classic platform games because these games don't use or require realistic physics. So the physics engine will be a choice, not something built-in.
Depending on that choice, area and body will be behave differently, but their API will stay largely the same.

**Comment by amyspark-ng** [comment]
could this be renamed as just adding physics system unless this is already being worked on
